src/Datadog.Trace.ClrProfiler.Native/clr_helpers.cpp

```diff
@@ -375,18 +375,17 @@ std::vector<IntegrationMethod> FilterIntegrationsByTarget(
 
   for (auto& i : integration_methods) {
     bool found = false;
-    if (AssemblyMeetsIntegrationRequirements(assembly_metadata,
-                                             i.replacement)) {
+    if (AssemblyMeetsIntegrationRequirements(assembly_metadata, i.replacement)) {
       found = true;
-    }
-    for (auto& assembly_ref : EnumAssemblyRefs(assembly_import)) {
-      const auto metadata_ref =
-          GetReferencedAssemblyMetadata(assembly_import, assembly_ref);
-      // Info(L"-- assembly ref: " , assembly_name , " to " , ref_name);
-      if (AssemblyMeetsIntegrationRequirements(metadata_ref, i.replacement)) {
-        found = true;
+    } else {
+      for (auto& assembly_ref : EnumAssemblyRefs(assembly_import)) {
```

Comment:
I know that the same behavior as before, but once AssemblyMeetsIntegrationRequirements returns true, we can break the loop ?
I don't know how time we are in this loop.

Reply:
done in 461a6c5